### PR TITLE
Create Ivadomed download function

### LINKED ISSUE #346: Create an ivadomed download function
instead of having developers and users use curl (e.g. in tutorial, for testing) as we currently advise, we should have an ivadomed downloader. Pros:
- currently, when unpacked, bundled artifacts on github have the name of the release as suffix, which is problematic (we need to manually rename them, creates confusion, etc.)
- having a downloader, we can maintain a database listing the different URLs, mirrors, etc. for each dataset

### PR DESCRIPTION
**Pull request by lrouhier**
Creation of download_data. This function will be used to download binary files from multiple URL. It is based on spinalcordtoolbox's download function. 

Done: 
* Script creation
* Addition of entrypoint

fix #346

### CODE REVIEW

File: ivadomed/scripts/download_data.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+    dict_url = {
+        "data_example_spinegeneric": ["https://github.com/ivadomed/data_example_spinegeneric/archive/master.zip"],
+        "data_microscopy_SEM_example": ["https://github.com/ivadomed/data_microscopy_SEM_example/archive/master.zip"],
+        "data-testing": ["https://github.com/ivadomed/data-testing/archive/master.zip"],
+        "t2_tumor": ["https://github.com/ivadomed/t2_tumor/archive/master.zip"],
+        "t2star_sc": ["https://github.com/ivadomed/t2star_sc/archive/master.zip"],
+        "mice_uqueensland_gm": ["https://github.com/ivadomed/mice_uqueensland_gm/archive/master.zip"],
+        "mice_uqueensland_sc": ["https://github.com/ivadomed/mice_uqueensland_sc/archive/master.zip"],
+        "findcord_tumor": ["https://github.com/ivadomed/findcord_tumor/archive/master.zip"]
```
Comment (jcohenadad) [inline ivadomed/scripts/download_data.py:214]: no, we should use the immutable artifacts generated by [distriploy](https://github.com/neuropoly/distriploy). E.g.: https://github.com/ivadomed/t2_tumor/releases/download/r20200621/r20200621_t2_tumor.zip

File: ivadomed/scripts/download_data.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    Args:
+        urls (list): List of urls to try.
+
+    Returns:
+        downloaded folder path
+    """
+    if isinstance(urls, str):
+        urls = [urls]
+
+        # loop through URLs
```
Comment (jcohenadad) [inline ivadomed/scripts/download_data.py:44]: PEP8

File: ivadomed/scripts/download_data.py (hunk 1 above)
Comment (jcohenadad) [reply ivadomed/scripts/download_data.py:214]: because of https://github.com/neuropoly/distriploy/issues/7 we can go ahead with a simple tag

Comment (lrouhier) [reply ivadomed/scripts/download_data.py]: Started in 3fafe8f. Winting on release from data-testing to finish up.

**Pull request merged**